integration-tests/test/test_slash.py

```diff
@@ -120,3 +120,47 @@ def test_slash_invalid_block_seq(command_line_options: CommandLineOptions, rando
         bonds_validators = extract_validator_stake_from_bonds_validator_str(block_info['bondsValidatorList'])
 
         assert bonds_validators[BONDED_VALIDATOR_KEY_1.get_public_key().to_hex()] == 0.0
+
+@pytest.mark.skipif(sys.platform in ('win32', 'cygwin', 'darwin'), reason="Only Linux docker support connection between host and container which node client needs")
+def test_slash_justification_not_correct(command_line_options: CommandLineOptions, random_generator: Random, docker_client: DockerClient) -> None:
+    """
+    Slash a validator which proposed a block with justifications not matching bonded validators
```

Comment:
Slash a validator which proposed a block with justifications not matching bonded validators of main parent